File: backend/models/article.go
package models

import "gopkg.in/mgo.v2/bson"

// Article model
type Article struct {
	ID       bson.ObjectId `json:"_id,omitempty" bson:"_id,omitempty"`
	Title    string        `json:"title" bson:"title"`
	Body     string        `json:"body" bson:"body"`
	User     User          `json:"user" bson:"user"`
	Comments []Comment     `json:"comments,omitempty" bson:"comments,omitempty"`
}

File: frontend/src/components/App/Main/Home/ArticleContainer/Article/Article.js
import React from 'react';
import Comment from './Comment/Comment';

const Article = (props) => {
    var commentArray = [];
    if (props.articleData.hasOwnProperty('comments')) {
        for (var i = 0; i < props.articleData.comments.length; i++) {
            commentArray.push(<Comment key={i} commentData={props.articleData.comments[i]} />);
        }
    }
    return (
        <div className="article" data-id={props.articleData._id}>
            <h1>{props.articleData.title}</h1>
            <p>{props.articleData.body}</p>
            <p data-id={props.articleData._id}>{props.articleData.user.username}</p>
            {commentArray}
        </div>
    )

}

export default Article;